### ISSUE TITLE
The testing methods can cause the program to Exit(1)

### ISSUE THREAD

**Issue by MEschenbacher** [showwin/speedtest-go#27]
Hey,

I've noticed that many of the helper functions for testing methods such as `{Upload,Download,Ping}Test` call `checkError(err)`. This function `Exit(1)`s the program if `err` is not `nil`.

I am using this library in a long running program and currently, a number of conditions out of the scope of the library and my program (e.g. network connectivity, DNS availability, even _misbehaving_ speedtest servers) may cause the library to terminate my program.

Are you open to discussing a pull request which changes this behavour?

**Comment by showwin** [comment]
Hi @MEschenbacher, thank you for using speedtest-go!
I'm open to discuss, of course :smile: 

I might not understand your situation completely, though if you use speedtest-go on cli mode in shell script, like:

```shell
#!/bin/bash

echo 'hello'
speedtest-go
echo 'world'
```

when speedtest-go fails, the result is like:
```
hello
Warning: Cannot fetch user information. http://www.speedtest.net/speedtest-config.php is temporarily unavailable.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x742b57]

goroutine 1 [running]:
main.showUser(0x0)
	/path/to/speedtest-go/speedtest.go:102 +0x37
main.main()
	/path/to/speedtest-go/speedtest.go:27 +0x7c
world
```

speedtest-go does not interfere following steps.

Are you using `set -e` option in your script? 
This option should terminate the script when `exit 1` is returned.

I'm not sure this answer is helpful for you :thought_balloon: .
If not, could you provide more information about your environment.

thanks,

**Comment by MEschenbacher** [comment]
I am using the "library" part/functions of this project because I am integrating the functionality in another project of mine.

This means I am not executing `speedtest-go` as binary but compiling the methods into my project. Whenever a method calls `Exit()` my program exits although I would rather like to get returned an `error` and deal with the situation myself.

**Comment by showwin** [comment]
I'm sorry for the misunderstanding :pensive: 
I fixed the issue at https://github.com/showwin/speedtest-go/pull/28, and released [v1.1.1](https://github.com/showwin/speedtest-go/releases/tag/v1.1.1) which includes this fix.

thanks,

**Comment by MEschenbacher** [comment]
Thank you for the fast response!
